File: Queue.py
# coding utf-8

class Queue:
  def __init__(self, *args):
      self.first = None
      self.last = None
      self.queue = []
      self.size = 0
      if args:
        self.first = args[0]
        for arg in args:
          self.add(arg)

  def add(self, *args):
    if args:
      self.last = args[-1]
      for value in args:
        self.queue.append(value)
        self.size += 1

  def remove(self):
    if self.size > 0:
      if len(self.queue) == 1:
        self.last = self.queue[0]
        self.first = self.queue[0]
      else:
        self.first = self.queue[1]
      self.size -= 1
      return self.queue.pop(0)
    self.last = None
    self.first = None
    self.size = 0
    return None

  def values(self):
    return self.queue

  def __str__(self):
      return 'Queue -> [{}] (size = {})'.format(', '.join([str(x) for x in self.queue]), self.size)

  def __len__(self):
      return self.size
"""
query = Queue(1, 2, 3, 4, 5, 0)
query.add(1,2)
print(query.first)
print(query.last)
print(query.size)
print(query)
print('-'*5)
print(query.remove())
print('-'*5)
print(query.first)
print(query.last)
print(query.size)
print(query)
for i in range(0,10):
  print(query.remove())
  print(query)
print(query.first)
print(query.last)
print(query.size)
print(query)
"""

File: Queue.js
class Queue {
  
  constructor(...props){
    this.queue = [];
    this.first = undefined;
    this.last = undefined;
    this.size = 0;
    if(props){
      this.first = props[0];
      this.last = props[props.length -1];
      for (const value of props) {
        this.add(value);
      }
    }
  }

  add(...values){
    this.last = values[values.length -1]
    for (const value of values) {
      this.queue.push(value);
      this.size += 1;
    }
  }

  remove(){
    this.first = this.queue[1];
    this.size -=1;
    return this.queue.shift();
  }

  toString(){
    return `Queue -> [${this.queue.join(', ')}] (size = ${this.size})`
  }
}

const queue = new Queue(1, 2, 3, 4, 5, 0);
queue.add(1,2)
console.log(queue.first)
console.log(queue.last)
console.log(queue.size)
console.log(queue.toString())
console.log('-----')
console.log(queue.remove())
console.log('-----')
console.log(queue.first)
console.log(queue.last)
console.log(queue.size)
console.log(queue.toString())